PT-2465 patch fix for erroneous flavour mismatch in mariadb-10.5.x and above

Nyele93 opened the pull request:
PT-2465 patch fix for erroneous flavour mismatch in mariadb-10.5.x and above. 

This patch includes the fixed regex for checking DB flavour to include the case-insensitive marker `i` on the searches for `MariaDB`

- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [x] util/update-modules has been ran
- [ ] Documentation updated
- [x] Test suite update

Nyele93 committed:
PT-2465 patch fix for erroneous flavour mismatch in mariadb-10.5.x and above

Nyele93 commented:
I have created a simple test case in . Below is the run of the test case: `t/pt-online-schema-change/mariadb_flavour_105_plus.t` , to test my patch against an **_ADD INDEX ALTER_** on MariaDB version -  **_10.6.22-18-MariaDB-enterprise_**
```
mamadi@mariadb-master: $ ./bin/pt-online-schema-change --version                                        ✔
pt-online-schema-change 3.7.0
mamadi@mariadb-master: $ prove -vw t/pt-online-schema-change/mariadb_flavour_105_plus.t                 ✔
t/pt-online-schema-change/mariadb_flavour_105_plus.t ..
1..3
# Creating test database and table...
# Running pt-online-schema-change with full options...
ok 1 - pt-online-schema-change executed successfully
ok 2 - No flavor mismatch or unsupported error
ok 3 - Index 'test_column_idx' created
ok
All tests successful.
Files=1, Tests=3,  1 wallclock secs ( 0.01 usr  0.01 sys +  0.14 cusr  0.07 csys =  0.23 CPU)
Result: PASS
```

Nyele93 committed:
created test case for patch

svetasmirnova commented:
Thank you for the patch. It still modifies auto-generated code, so your changes would be overwritten next time we make a new release. Please follow the contribution guide in the file CONTRIBUTION.md to implement changes properly. Do not forget to run ./utils/update-modules to update other tools.

svetasmirnova commented:
Btw, we also have PR-965 with the same issues as yours.

Nyele93 committed:
executed ./util/update-modules bin/pt-online-schema-change

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=981) <br/>All committers have signed the CLA.

Nyele93 committed:
Merge branch 'percona:3.x' into PT-2465_patch_fix_for_mariadb_10.5_above

Nyele93 committed:
PT-2465 patch fix for erroneous flavour mismatch in mariadb-10.5.x and above. This patch includes the fixed regex for checking DB flavour to include the case-insensitive marker i on the searches for MariaDB

Nyele93 commented:
Thanks Sveta. I believe have followed the right process now, PR is updated.

svetasmirnova committed:
PT-2440 - Support for MariaDB 10.5

- Updated non-libraries code in bin/pt-heartbeat, bin/pt-replica-restart, and bin/pt-slave-delay
- Adjusted lib/Sandbox.pm, so it does not require SSL for MariaDB
- Adjusted test case, so it is in line with other tests

svetasmirnova commented:
Thank you for the update. I modified the test case to align with other tests and fixed missed checks for MariaDB in other tools. I will run the full test suite and likely include this PR in the next release.

svetasmirnova committed:
PT-2440 - Support for MariaDB 10.5

- Added sandbox defaults for MariaDB 11.3
